### ISSUE TITLE
Downgrade file to 6.1.4

### ISSUE THREAD

**Issue by fotiDim** [onepub-dev/dcli#222]
**Describe the bug**

I am using a dev dependency which depends on dcli. Resolving dependency versions fails:
```
Because ... depends on dcli ^3.0.3 which depends on file ^7.0.0...
And because every version of integration_test from sdk depends on file 6.1.4, ... is incompatible with integration_test from sdk.
```

**To Reproduce**
```yaml
dev_dependencies:
  flutter_test:
    sdk: flutter
  integration_test:
    sdk: flutter
  flutter_translation_sheet: ^1.0.30
```

**Expected behavior**
Using latest Flutter (3.13.1 as of now) version resolution should succeed. Downgrading `file` to `6.1.4` to make it compatible with `integration_test` should do the trick. 

⚠️ Also please make a release after that. The last public version is 3.0.3 which uses `pubspec2: ^2.5.0` which is another cause of dependency resolution failures.

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.1, on macOS 13.5 22G74 darwin-arm64, locale en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.
[✓] Android Studio (version 2022.2)
[✓] VS Code (version 1.81.1)
[✓] Connected device (2 available)
    ! Error: Browsing on the local area network for Eleni’s iPhone 12 Mini. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[✓] Network resources

! Doctor found issues in 1 category.
```

**Comment by bsutton** [comment]
I'm working through a couple of issues with the pubspec2 package, once that is resolved I will be able to release a fixed version of dcli.

**Comment by bsutton** [comment]
Can you try the latest release of dcli - 3.1.0 and see if it resolves your issue.

**Comment by fotiDim** [comment]
@bsutton partially only. You haven't downgraded file to 6.1.4 so it still conflicts with integration_test.

**Comment by bsutton** [comment]
So the problem is that dcli uses posix which uses quiver which apparently depends on matcher  and hence the test_api.

I will see if I can change posix to no use quiver.

**Comment by fotiDim** [comment]
@bsutton made a small PR which should address the issue.

**Comment by bsutton** [comment]
I've just released 3.2.1 which should resolve your issue.
